src/st/no0/e_marionette.c

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+#include "common.h"
+#include "no0.h"
+
+u8 func_us_801D6254(s16 arg0[]) {
+    Collider collider;
+    s16 xDir;
+    u8 ret = 0;
+    s32 i;
+    s16 posX;
+    s16 posX2; // Possible !FAKE here
+    s16 posY;
+    s16 posY2;
+
+    g_CurrentEntity->posX.val += g_CurrentEntity->velocityX;
+    g_CurrentEntity->posY.val += g_CurrentEntity->velocityY;
+    posX2 = g_CurrentEntity->posX.i.hi;
+    posY = g_CurrentEntity->posY.i.hi + *arg0++;
+    posY2 = g_CurrentEntity->posY.i.hi + *arg0++;
+
+    if (g_CurrentEntity->velocityX > 0) {
+        xDir = *arg0;
+    } else {
+        xDir = -*arg0;
+    }
+
+    posX = posX2; // Possible !FAKE here
```

Comment:
This stuff seems a bit unintuitive to me so may be a fake variable here, or just some strangely written code

Reply:
Since it's matching on psp it's probably real

Reply:
Done!